app/models/g5_authenticatable/user.rb

```diff
@@ -38,7 +38,7 @@ def update_roles_from_auth(auth_data)
     end
 
     def clients
-      G5Updatable::ClientPolicy::Scope.new(self, G5Updatable::Client).resolve
+      G5Updatable::ClientPolicy::Scope.new(self, G5Updatable::Client).clients_from_client_and_location_roles
```

Comment:
That is just the thing. Having access to a location does not give you
permission at the client level. For example, if you a client ("Pebble
Beach") and have an admin role at location "Spanish Bay", in the dashboard
you will only be able to see the "Spanish Bay" view of any report and not
the "All Locations" view. Make sense?

On Tue, Feb 16, 2016 at 10:51 AM, Victor Garro <notifications@github.com>
wrote:

> In app/models/g5_authenticatable/user.rb
> <https://github.com/G5/g5_authenticatable/pull/46#discussion_r53058290>:
>
> > @@ -38,7 +38,7 @@ def update_roles_from_auth(auth_data)
> >      end
> >
> >      def clients
> > -      G5Updatable::ClientPolicy::Scope.new(self, G5Updatable::Client).resolve
> > +      G5Updatable::ClientPolicy::Scope.new(self, G5Updatable::Client).clients_from_client_and_location_roles
>
> I think that an indirect role (location) should entitle the user to it's
> parent role (client) on all the cases, and not on selected ones. If there's
> a case for "just listing the direct client roles", then it makes sense to
> have an specialized method to do so, but not the other way around since
> it's a primary use case. I'll let @maeve <https://github.com/maeve> be
> our judge :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/G5/g5_authenticatable/pull/46/files#r53058290>.
>



-- 
Jonathan